sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -126,25 +148,114 @@
 
   @Override
   public Span startSpan() {
-    // TODO: get remoteParent span from the context if noParent=false and remoteParent/remoteParents
-    // are null
-    // TODO: correctly implement this.
-    SpanContext context =
-        SpanContext.create(
-            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, 0),
-            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8}, 0),
-            TraceOptions.builder().setIsSampled(true).build(),
-            Tracestate.getDefault());
+    Span currentSpan = null;
+    if (parentType == ParentType.CURRENT_SPAN) {
+      currentSpan = ContextUtils.getValue();
+    }
+    SpanContext parentContext = parent(parentType, currentSpan, parent, remoteParent);
+    SpanContext spanContext = createSpanContext(spanName, parentContext, sampler);
+    boolean recordEvents =
```

Comment:
The recordEvents logic is that it is true if this Span records events, which we do if the span is sampled OR user especially set recordEvents.
```
recordEvents = this.recordEvents || spanContext.getTraceOptions().isSampled();
```

The other thing is that recordEvents in the builder can be a `boolean` with a default of `false`.

Reply:
I have removed this for now. As I understand it `DefaultSpan` will be returned when not recording events, however `DefaultSpan` cannot be created with supplied span context. I would prefer to handle this in a separate issue.

Reply:
https://github.com/open-telemetry/opentelemetry-java/pull/430#discussion_r296911812

Reply:
It will be done in https://github.com/open-telemetry/opentelemetry-java/issues/438